sdk/src/main/java/io/customer/sdk/queue/type/QueueTaskGroup.kt

```diff
@@ -0,0 +1,19 @@
+package io.customer.sdk.queue.type
+
+/**
+ * All of the different groups that are possible in the queue.
+ *
+ * Not sure what queue groups are? See the `BACKGROUND_QUEUE.md` doc.
+ */
+sealed class QueueTaskGroup {
+    /**
+     * Be sure that the string is unique to each subclass of [QueueTaskGroup].
+     * The returned string will identify each group in the background queue. Therefore, it's important to use properties from the subclass constructor in the returned string to differentiate groups of the same type.
+     */
+    data class IdentifyProfile(val identifier: String) : QueueTaskGroup() {
+        override fun toString(): String = "identified_profile_$identifier"
+    }
+    data class RegisterPushToken(val token: String) : QueueTaskGroup() {
+        override fun toString(): String = "registered_push_token_$token"
+    }
```

Comment:
I really like the grouping feature, there are a lot of use cases for it, not just pre-defined but maybe runtime as well. 

For example, there is a particular event we want to track but we want some other events to get tracked before it, a real-life example could be, I create an event `Onboarding Completed` but then I block it by the group, `Onboarding`. And when a user is going through each onboarding page, I am adding the event with the group `Onboarding"` in it. Once the user goes through all onboarding pages, and the `Onboarding` group is completed, the previously blocked event `Onboarding Completed` gets sent as well. 

 but if we only accept `QueueTaskGroup` at QueueRunner, we then need another subclass that lets us create `QueueTaskGroup` subclasses. 
 
Thoughts?

Reply:
🤔 I have never thought of using groups at runtime before. I have only thought about the specific use case it's being used in today for helping ensure that HTTP events will run successfully in the queue. 

With the real-world example that you gave, I believe that you are suggesting that *events* are sent to CIO in a specific order, correct? 

If you don't mind, I want to ask product to see if this is a use case that is valid for customers. I asked in our slack channel. 

Good question!

Reply:
Added a bit more detail on it in the thread, just a *note* it's not a blocker, so this suggestion can be ignored and can be taken on later.

Reply:
Let's continue the conversation outside GitHub. I'll close this.